### PR TITLE
feat: add docs for improved account linking

### PR DESCRIPTION
**Pull request by jonas-jonas**
## Checklist

- [ ] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md) and signed the CLA.
- [ ] I have referenced an issue containing the design document if my change introduces a new feature.
- [ ] I have read the [security policy](../security/policy).
- [ ] I confirm that this pull request does not address a security vulnerability. 
      If this pull request addresses a security vulnerability, 
      I confirm that I got approval (please contact [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push the changes.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added the necessary documentation within the code base (if appropriate).

### CODE REVIEW

File: docs/kratos/social-signin/95_account-linking.mdx [hunk 1]
```diff
@@ -67,5 +93,13 @@ Automatic account linking can be a security risk. Consider this scenario:
 1. Your system, detecting duplicate accounts, prompts for account linking.
 1. Malicious actors link the accounts, gaining access to John's account.
 
-To prevent this users need to verify an additional credential before the accounts can be linked. In the scenario above, the
+To prevent this, users need to verify an additional credential before the accounts can be linked. In the scenario above, the
 malicious actors would be prompted to enter the password associated with the `jon@doe.com` identifier.
+
+### Hiding irrelevant authentication methods
```
Comment (vinckr) [inline docs/kratos/social-signin/95_account-linking.mdx:99]:
```suggestion
### Hide irrelevant authentication methods
```

**Review by vinckr — COMMENTED** [review]
lgtm, small nit

File: docs/kratos/social-signin/95_account-linking.mdx [hunk 2]
```diff
@@ -67,5 +93,13 @@ Automatic account linking can be a security risk. Consider this scenario:
...
+To prevent this, users need to verify an additional credential before the accounts can be linked. In the scenario above, the
 malicious actors would be prompted to enter the password associated with the `jon@doe.com` identifier.
+
+### Hide irrelevant authentication methods
+
+Due to security considerations, it might be required to show all configured authentication methods to the user. However, this can
+be confusing for users who have multiple accounts with different authentication methods.
+
+To hide irrelevant authentication methods, you can use the [login hints](../../identities/sign-in/login-hint.mdx) feature. By
+enabling login hints, users are informed about the authentication methods available for the existing account.
```
Comment (vinckr) [inline docs/kratos/social-signin/95_account-linking.mdx:105]: IMO a bit confusing what this means exactly. 
So when you enable login hints it hides the authentication method that the user selected in the first step (the method that is _not_ linked to their identity)?

File: docs/kratos/social-signin/95_account-linking.mdx [hunk 3]
```diff
@@ -67,5 +93,13 @@ Automatic account linking can be a security risk. Consider this scenario:
 1. Your system, detecting duplicate accounts, prompts for account linking.
 1. Malicious actors link the accounts, gaining access to John's account.
 
-To prevent this users need to verify an additional credential before the accounts can be linked. In the scenario above, the
+To prevent this, users need to verify an additional credential before the accounts can be linked. In the scenario above, the
 malicious actors would be prompted to enter the password associated with the `jon@doe.com` identifier.
+
+### Hide irrelevant authentication methods
```
Comment (vinckr) [inline docs/kratos/social-signin/95_account-linking.mdx:99]: Which methods do we hide specifically, or what makes them irrelevant? 
hard to tell from the headline what this means

**Comment by aeneasr** [comment]
What can be done to get this over the line?

**Comment by jonas-jonas** [comment]
I need to address the comments...

**Review by vinckr — APPROVED** [review]

**Pull request merged**